lib/features/modeling/BpmnUpdater.js

```diff
@@ -171,6 +179,49 @@ export default function BpmnUpdater(
     }
   });
 
+  // Handle embedded labels with defined bounds, to stay compatible with other vendors
+  this.executed('shape.move', function(event) {
+    var shape = event.context.shape;
+
+    if (hasExplicitlyBoundedEmbeddedLabel(shape)) {
```

Comment:
If we hook in prior to the DI update we#ll be able to diff bounds (shape bounds vs. bounds stored in DI).

No need for separate logic to handle shape move (do / undo) as well as resize (do / undo).